src/main/java/ch/uzh/ifi/hase/soprafs23/service/AuthenticationService.java

```diff
@@ -28,6 +28,16 @@ public AuthenticationService(@Qualifier("userRepository") UserRepository userRep
 
     public User authenticateUser(User externalUser) {
         User user = this.userRepository.findByUsername(externalUser.getUsername());
+        if (externalUser.getPassword().equals("") || externalUser.getUsername().equals("")) {
+            String baseErrorMessage = "Oups, your request is wrong. Please full out all fields!";
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+                    String.format(baseErrorMessage));
+        }
+        if (externalUser.getPassword().length() > 18 || externalUser.getUsername().length() > 18) {
+            String baseErrorMessage = "Please don't use more than 20 characters for username or password!";
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+                    String.format(baseErrorMessage));
+        }
```

Comment:
die Methode war eig ok so wie sie war

Reply:
ja aber wenn jemand einen username mit 100 Zeichen nimmt sieht es scheisse im frontend aus

Reply:
authenticateUser wir nur bei login abgerufen, bei login is der user name ja schon fix, also kein grund noch mehr zu überprüfen

Reply:
gelöscht